pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1194,6 +1206,66 @@ func (p *PravegaCluster) validateConfigMap() error {
 	return nil
 }
 
+func (p *PravegaCluster) ValidateSegmentStore() error {
+	totalMemoryLimitsQuantity := p.Spec.Pravega.SegmentStoreResources.Limits[corev1.ResourceMemory]
+	totalMemoryRequestsQuantity := p.Spec.Pravega.SegmentStoreResources.Requests[corev1.ResourceMemory]
```

Comment:
How are you verifying the scenario, limit is set but requests are not set for resources

Reply:
Setting the request to the limit value should be done in the defaults section, if I am not mistaken.

@nishant-yt Did you verify that `totalMemoryRequestsQuantity.Value()` below produces a zero when the memory request is NOT set by the user?

Reply:
Added the code in defaults section to make sure requests is equal to limits in case requests is not set by the user. Also,  I have verified `totalMemoryRequestsQuantity.Value()` does produces `0` value if memory requests is not set by the user